### DISCUSSION TITLE
Beta testing and "breaking" changes

### DISCUSSION THREAD

**Discussion by technicallypete** [nuxt/nuxt#18015] [General]
**Context:** I am building a product on Nuxt3 already, so that when Nuxt3 is ready, so is my thing...

Take away from that experience: _if_ Nuxt is in beta, and people like myself are essentially the beta testers (trying to already build real-world products on Nuxt3), then I would like to propose that developers have a _little_ more grace in acknowledging this, and all I'm asking for is that for any possible "breaking" beta change **please** post an appropriate announcement in Discussions.

**Comment by tobiasdiez** [comment]
Another approach, that might be easier for both sides, would be to introduce version numbers and a changelog (highlighting breaking changes). 
This would maybe also solve the issue that one sometimes have to delete the lock file in order to fully upgrade to the new version.

**Reply by pi0** [reply]
Certainly! Starting with rc releaes soon, we have manual releases with complete changelog and potential breaking changes to be considered (however should happen less often by then). `nuxt-edge` and `nuxt3` are auto released from main branch mainly to try features before final release.

**Comment by pi0** [comment]
We are trying hard to finalize API changes of Nuxt3 and sometimes breaking changes are inevitable until reaching 3.0.0 stable. Please consider following repository for the issue and PR notifications and switching to rc releases that will be announced with release notes.
